bootstrap/kubeadm/api/v1alpha3/conversion.go

```diff
@@ -51,6 +51,10 @@ func (src *KubeadmConfig) ConvertTo(dstRaw conversion.Hub) error {
 		dst.Spec.InitConfiguration.NodeRegistration.IgnorePreflightErrors = restored.Spec.InitConfiguration.NodeRegistration.IgnorePreflightErrors
 	}
 
+	if dst.Spec.Ignition == nil && restored.Spec.Ignition != nil {
+		dst.Spec.Ignition = restored.Spec.Ignition
+	}
```

Comment:
```suggestion
	dst.Spec.Ignition = restored.Spec.Ignition
```
Shouldn't we always overwrite in this case given that the field doesn't exist in previous versions?

Reply:
What if the controller is older and someone submits an older configuration? This way we don't lose the config since it will be extracted from the annotations, isn't it?

Reply:
Ah sorry my fault. I came to the wrong conclusion in Slack. I think it's fine to always set it. The reason is that it will always be nil in dst as dst is converted from src (which doesn't have the field)

Reply:
I was [discussing](https://kubernetes.slack.com/archives/C8TSNPY4T/p1632922077076000?thread_ts=1632902667.064500&cid=C8TSNPY4T) this exact thing on slack with Stefan.

Reply:
Yup, I just think my response here was wrong: https://kubernetes.slack.com/archives/C8TSNPY4T/p1632922351077000?thread_ts=1632902667.064500&cid=C8TSNPY4T

Because: 
>  I think it's fine to always set it. The reason is that it will always be nil in dst as dst is converted from src (which doesn't have the field)

Reply:
Done!